jupyterlab_comments/jupyterlab_comments/handlers_test.py

```diff
@@ -0,0 +1,233 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import os
+import tempfile
+import handlers
+import subprocess
+import unittest
+import json
+import tornado.ioloop
+import tornado.web
+import tornado.testing
+
+remote_dir = tempfile.TemporaryDirectory()
+not_git_dir = tempfile.TemporaryDirectory()
+dir1 = tempfile.TemporaryDirectory()
+dir2 = tempfile.TemporaryDirectory()
+dir1_path = dir1.name
+dir2_path = dir2.name
+
+
+home = os.path.expanduser("~")
+FNULL = open(os.devnull, 'w')
+
+def create_files(num_files, dir_name):
+    for i in range(1, num_files + 1):
+        with open(os.path.join(dir_name, "test" + str(i) + ".py"), 'a') as temp_file:
+            temp_file.write("# Test file")
+
+def create_tornado_app():
+    app = tornado.web.Application()
+    host_pattern = '.*$'
+    app.add_handlers(host_pattern, [
+        ('/detachedComments', handlers.DetachedCommentsHandler),
+        ('/remotePull', handlers.PullFromRemoteRepoHandler),
+        ('/addDetachedComment', handlers.AddDetachedCommentHandler),
+        ('/reviewComments', handlers.ReviewCommentsHandler)])
+    return app
+
+def setup_repos():
```

Comment:
I would suggest passing in the current working directory as an argument (which could be a temp directory), and then creating all of the repo dirs as subdirectories of that.

That way, you can have a different set of repositories used by each test.

I would additionally suggest returning the names of the repository directories so that they don't have to be hard-coded.

That would be easiest if you defined an object that represented a test environment, made each of the created repository names a field in that object, and then returned the object.

Something like:

```py

class TestEnv(object):
    def __init__(self, remote, clone1, clone2):
        self.remote = remote
        self.clone1 = clone1
        self.clone2 = clone2

def setup_repos(cwd):
   ...
   return TestEnv(remote_dir, dir1, dir2)
```